internal/generator/vector/output/syslog/syslog.go

```diff
@@ -193,18 +193,22 @@ func getEncodingTemplatesAndFields(s *obs.Syslog) EncodingTemplateField {
 }
 
 func Encoding(id string, o obs.OutputSpec, templatePairs []FieldVRLStringPair) []Element {
-	// Set the appropriate encoding fields that have been user templated
-	encodingFields := []Element{
-		SyslogEncoding{
-			ComponentID:  id,
-			RFC:          strings.ToLower(string(o.Syslog.RFC)),
-			Facility:     Facility(o.Syslog),
-			Severity:     Severity(o.Syslog),
-			AddLogSource: genhelper.NewOptionalPair("add_log_source", o.Syslog.Enrichment == obs.EnrichmentTypeKubernetesMinimal),
-			PayloadKey:   genhelper.NewOptionalPair("payload_key", "payload_key"),
-		},
+	sysLEncode := SyslogEncoding{
+		ComponentID:  id,
+		RFC:          strings.ToLower(string(o.Syslog.RFC)),
+		Facility:     Facility(o.Syslog),
+		Severity:     Severity(o.Syslog),
+		AddLogSource: genhelper.NewOptionalPair("add_log_source", o.Syslog.Enrichment == obs.EnrichmentTypeKubernetesMinimal),
+		PayloadKey:   genhelper.NewOptionalPair("payload_key", nil),
+	}
+	if o.Syslog.PayloadKey != "" {
+		sysLEncode.PayloadKey.Value = "payload_key"
```

Comment:
I would look at optional pair and what it does when it results in a string but this logic can either be:
```
genhelper.NewOptionalPair("payload_key", o.Syslog.PayloadKey)
```
or  must be:
```
if o.Syslog.PayloadKey != "" {
		sysLEncode.PayloadKey.Value = o.Syslog.PayloadKey
```
you don't want it to be the literal 'payload_key' right?